Add CSS color variables to Card component

tagliala opened the pull request:
Hi, this is an attempt to provide variables for the Card component, but I have a couple of questions

- `$card-color` and `$card-cap-color` are defined as `null`. This has the effect of rules being generated with SCSS, but it will generate empty css variables.
  - Is it fine to have empty CSS variables?
  - Should the default value change to `inherit`?
  - if the default value change from `null` to `inherit`, should it be considered a breaking change?
2. Should the conditional rules in `@if $nav-tabs-link-active-bg != $card-bg ` moved out from `if`?

mdo commented:
@tagliala I meant something like this:

```scss
cursor: if($enable-button-pointers, pointer, null);
```

That said, it _might_ be fine to have some unused variables lying around for now provided we have a documented approach for them. Unused or `null` variables in Sass exist in case someone wants to customize. I think given our hybrid CSS vars / Sass solution in v5, we can keep this going, I'm just not locked in on any one direction for it.

tagliala committed:
Add CSS variables

tagliala committed:
Remove conditional rules

tagliala committed:
Fix calculations

tagliala committed:
Update docs

tagliala commented:
Hello, I've rebased this PR on master and I've replaced all card scss variables into css variables

About the definitions, I've used the same order of scss variables, but I know that the css is different

The conversion is based on an automatic find and replace to prevent typos

Feel free to take over this PR

mdo commented:
Closed for #36059.
